src/CommonProviderImplementation/Helpers.fs

```diff
@@ -171,9 +172,10 @@ module internal ProviderHelpers =
         using (logTime "Loading" sampleOrSampleUri) <| fun _ ->
     
         let tryGetResource() = 
```

Comment:
tryGetResources is called right after this, and only once, what do you gain with the Lazy?